### PR TITLE
Add view chains handler

### PR DESCRIPTION
**Pull request by esuwu**

### CODE REVIEW

**Comment by esuwu** [comment]
TODO: remove the height difference limitation (10)

**Comment by esuwu** [comment]
TODO: double-check the behavior with the special nodes

File: pkg/api/api.go [hunk 1]
```diff
@@ -196,13 +196,16 @@ func (a *API) specificNodesHandler(w http.ResponseWriter, r *http.Request) {
...
 	// TODO: these nodes don't send base target value at the moment
+	// TODO: these nodes don't send a generator at the moment
 	stateHashEvent := entities.NewStateHashEvent(
 		statement.Node,
 		zeroTS,
 		statement.Version,
 		statement.Height,
 		statehash,
 		zeroBT,
+		statehash.BlockID,
```
Comment (nickeskov) [inline pkg/api/api.go:207]: I think the `BlockID` emptiness check should be added before using the field.

File: cmd/bots/internal/telegram/handlers/handlers.go [hunk 2]
```diff
@@ -158,6 +160,39 @@ func onTextMsgHandler(
...
+func viewChains(
+	requestChan chan<- pair.Request,
+	responsePairType <-chan pair.Response,
+	ext common.ExpectedExtension,
+	zapLogger *zap.Logger,
+) func(c telebot.Context) error {
+	return func(c telebot.Context) error {
+		nodes, err := messaging.RequestAllNodes(requestChan, responsePairType)
+		if err != nil {
+			zapLogger.Error("failed to get nodes list", zap.Error(err))
```
Comment (nickeskov) [inline cmd/bots/internal/telegram/handlers/handlers.go:172]: Should `err` be returned and sent through response chan?

**Review by nickeskov — CHANGES_REQUESTED** [review]

File: pkg/api/api.go (hunk 1 above)
Comment (esuwu) [reply pkg/api/api.go]: Here it is not a pointer

File: cmd/bots/internal/telegram/handlers/handlers.go (hunk 2 above)
Comment (nickeskov) [reply cmd/bots/internal/telegram/handlers/handlers.go:178]: Guess not, okay

**Review by nickeskov — APPROVED** [review]

**Comment by nickeskov** [comment]
@esuwu is [this TODO](https://github.com/wavesplatform/nodemon/pull/225#issuecomment-1968545046) fixed?

File: pkg/api/api.go [hunk 3]
```diff
@@ -196,13 +196,17 @@ func (a *API) specificNodesHandler(w http.ResponseWriter, r *http.Request) {
...
+	// TODO: support the generator field
 	stateHashEvent := entities.NewStateHashEvent(
 		statement.Node,
 		zeroTS,
 		statement.Version,
 		statement.Height,
 		statehash,
 		zeroBT,
+		&statehash.BlockID,
+		nil,
```
Comment (esuwu) [inline pkg/api/api.go:209]: If privates nodes are not going to send us the generator info, let's put an empty address here instead of nil

**Review by alexeykiselev — APPROVED** [review]

**Pull request merged**